File: Algorithms/C#/crossplatform/ClassLibrary1/PairNode.cs
using System;
namespace algorithms
{
	
	/// <summary> Public class for use with PairHeap. It is public
	/// only to allow references to be sent to decreaseKey.
	/// It has no public methods or members.
	/// </summary>
	/// <author>  Mark Allen Weiss, Jonathan Moore
	/// </author>
	/// <seealso cref="PairHeap">
	/// </seealso>
	public class PairNode
	{
		/// <summary> Construct the PairNode.</summary>
		/// <param name="theElement">the value stored in the node.
		/// </param>
		internal PairNode(Comparable theElement)
		{
			element = theElement;
			leftChild = null;
			nextSibling = null;
			prev = null;
		}
		
		// Friendly data; accessible by other package routines
		internal Comparable element;
		internal PairNode leftChild;
		internal PairNode nextSibling;
		internal PairNode prev;
	}
}

File: Algorithms/Java/src/src/com/unsorted/PairNode.java
    package com.algorithms;

    /**
     * Public class for use with PairHeap. It is public
     * only to allow references to be sent to decreaseKey.
     * It has no public methods or members.
     * @author Mark Allen Weiss, Jonathan Moore
     * @see PairHeap
     */
    public class PairNode
    {
        /**
         * Construct the PairNode.
         * @param theElement the value stored in the node.
         */
        PairNode( Comparable theElement )
        {
            element     = theElement;
            leftChild   = null;
            nextSibling = null;
            prev        = null;
        }

            // Friendly data; accessible by other package routines
        Comparable element;
        PairNode   leftChild;
        PairNode   nextSibling;
        PairNode   prev;
    }
